internal/eru/agent/workload.go

```diff
@@ -0,0 +1,171 @@
+		ids, err = m.svc.GetGuestIDList(ctx)
+	})
+	if err != nil && !strings.Contains(err.Error(), "key not exists") {
+		log.WithFunc("ListWorkloadIDs").Error(ctx, err, "failed to get workload ids")
+		return nil, err
+	}
+	return ids, nil
+}
+
+func (m *Manager) detectWorkload(ctx context.Context, ID string) (*Guest, error) {
+	logger := log.WithFunc("detectWorkload").WithField("ID", ID)
+
+	var guest *yavirttypes.Guest
+	var err error
+
+	utils.WithTimeout(ctx, m.config.GlobalConnectionTimeout, func(ctx context.Context) {
+		guest, err = m.svc.GetGuest(ctx, ID)
+	})
+
+	if err != nil {
+		logger.Error(ctx, err, "failed to detect workload")
+		return nil, err
+	}
+
+	if _, ok := guest.Labels[cluster.ERUMark]; !ok {
+		return nil, common.ErrInvaildVM
+	}
+
+	if m.config.CheckOnlyMine && m.config.Hostname != guest.Hostname {
+		logger.Debugf(ctx, "guest's hostname is %s instead of %s", guest.Hostname, m.config.Hostname)
+		return nil, common.ErrInvaildVM
+	}
+
+	return &Guest{
+		ID:            guest.ID,
+		Status:        guest.Status,
+		TransitStatus: guest.TransitStatus,
+		CreateTime:    guest.CreateTime,
+		TransitTime:   guest.TransitTime,
+		UpdateTime:    guest.UpdateTime,
+		CPU:           guest.CPU,
+		Mem:           guest.Mem,
+		Storage:       guest.Storage,
+		ImageID:       guest.ImageID,
+		ImageName:     guest.ImageName,
+		ImageUser:     guest.ImageUser,
+		Networks:      guest.Networks,
+		Labels:        guest.Labels,
+		IPs:           guest.IPs,
+		Hostname:      guest.Hostname,
+		Running:       guest.Running,
+		once:          sync.Once{},
+	}, nil
+}
+
+// GetStatus checks workload's status first, then returns workload status
+func (m *Manager) GetStatus(ctx context.Context, ID string, checkHealth bool) (*types.WorkloadStatus, error) {
+	logger := log.WithFunc("GetStatus").WithField("ID", ID)
+	guest, err := m.detectWorkload(ctx, ID)
+	if err != nil {
+		logger.Error(ctx, err, "failed to get guest status")
+		return nil, err
+	}
+
+	bytes, err := json.Marshal(guest.Labels)
+	if err != nil {
+		logger.Error(ctx, err, "failed to marshal labels")
+		return nil, err
+	}
+
+	status := &types.WorkloadStatus{
+		ID:        guest.ID,
+		Running:   guest.Running,
+		Healthy:   guest.Running && guest.HealthCheck == nil,
+		Networks:  guest.Networks,
+		Extension: bytes,
+		Nodename:  m.config.Hostname,
+	}
+
+	if checkHealth && guest.Running {
+		free, acquired := m.cas.Acquire(guest.ID)
+		if !acquired {
+			return nil, common.ErrGetLockFailed
+		}
+		defer free()
+		timeout := time.Duration(m.config.HealthCheck.Timeout) * time.Second
+		status.Healthy = guest.CheckHealth(ctx, m.svc, timeout, m.config.HealthCheck.EnableDefaultChecker)
+	}
+
+	return status, nil
+}
+
+func (m *Manager) healthCheck(ctx context.Context) {
+	tick := time.NewTicker(time.Duration(m.config.HealthCheck.Interval) * time.Second)
+	defer tick.Stop()
+
+	_ = utils.Pool.Submit(func() { m.checkAllWorkloads(ctx) })
```

Comment:
can remove `_ =` part from the statement

Reply:
removing `_=` will cause a lint issue